parser/parser.go

```diff
@@ -1075,46 +1077,86 @@ func (p *Parser) parseFormatStringOperator() (token.Token, string, string, error
+			cursorOverlapWidth = int(num)
+		},
+	}
+
+	for !p.peekTokenIs(token.RPAREN) {
+		if !p.peekTokenIs(token.COMMA) {
+			return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() parameter '%s'. Expected a comma", p.peekToken.Literal))
+		}
 
-	if p.peekTokenIs(token.COMMA) {
 		p.nextToken()
-		if p.peekTokenIs(token.STRING) {
+
+		if p.peekTokenIs(token.INT) {
+			num, _ := strconv.ParseInt(p.peekToken.Literal, 0, 64)
+			maxTextLength = int(num)
 			p.nextToken()
-			fontID = p.curToken.Literal
-			fontIdToken = p.curToken
-			if p.peekTokenIs(token.COMMA) {
-				p.nextToken()
-				if err := p.expectPeek(token.INT); err != nil {
-					return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() maxLineLength '%s'. Expected integer", p.peekToken.Literal))
-				}
-				num, _ := strconv.ParseInt(p.curToken.Literal, 0, 64)
-				maxTextLength = int(num)
-			}
-		} else if p.peekTokenIs(token.INT) {
+			continue
+		}
+
+		paramName := p.peekToken.Literal
+		_, ok := namedParameters[paramName]
+		if !ok {
+			fontID = p.peekToken.Literal
+			fontIdToken = p.peekToken
 			p.nextToken()
-			num, _ := strconv.ParseInt(p.curToken.Literal, 0, 64)
-			maxTextLength = int(num)
-			if p.peekTokenIs(token.COMMA) {
-				p.nextToken()
-				if err := p.expectPeek(token.STRING); err != nil {
-					return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() fontId '%s'. Expected string", p.peekToken.Literal))
-				}
-				fontID = p.curToken.Literal
-				fontIdToken = p.curToken
-			}
-		} else {
-			return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() parameter '%s'. Expected either fontId (string) or maxLineLength (integer)", p.peekToken.Literal))
+			continue
 		}
+
+		p.nextToken()
+
+		if !p.peekTokenIs(token.ASSIGN) {
+			return token.Token{}, "", "", NewParseError(p.peekToken, fmt.Sprintf("invalid format() parameter '%s'. Expected an equals sign", p.peekToken.Literal))
+		}
+
+		p.nextToken()
+		paramValue = p.peekToken.Literal
+
+		setNamedParameters, ok := namedParameters[paramName]
+
+		if !ok {
+			return token.Token{}, "", "", NewParseError(p.curToken, fmt.Sprintf("invalid format() parameter '%s'. Expected one of the following: fontId (string), maxLineLength (integer), cursorOverlapWidth (integer), numLines (integer).", paramName))
+		}
+		setNamedParameters()
+		p.nextToken()
 	}
 	if err := p.expectPeek(token.RPAREN); err != nil {
-		return token.Token{}, "", "", NewParseError(p.peekToken, "missing closing parenthesis ')' for format()")
+		return token.Token{}, "", "", NewParseError(p.curToken, "missing closing parenthesis ')' for format()")
 	}
 
 	if maxTextLength <= 0 {
 		maxTextLength = p.fonts.Fonts[fontID].MaxLineLength
 	}
 
-	formatted, err := p.fonts.FormatText(textToken.Literal, maxTextLength, p.fonts.Fonts[fontID].CursorOverlapWidth, fontID)
+	if numLines <= 0 {
+		numLines = p.fonts.Fonts[fontID].NumLines
+	}
```

Comment:
Maybe we should raise an error earlier where the `numLines` value is set, instead of silently changing it to the font config's value.

Reply:
https://github.com/huderlem/poryscript/pull/59/commits/667acc2e3be67c09bf4e0d52f3f7bb3083b785e2

this fires off a warning every time the user compiles and doesn't define the scripts. I commented it out, but either I misinterpreted the ask or this is actually a bad idea